src/DotNetWorker.Core/Hosting/CoreWorkerHostBuilderExtensions.cs

```diff
@@ -38,5 +40,48 @@ public static IHostBuilder ConfigureFunctionsWorker(this IHostBuilder builder, A
 
             return builder;
         }
+
+        /// <summary>
+        /// Invokes auto-generated configure methods for a given <see cref="IHostBuilder"/> instance.
+        /// This method searches for extension methods decorated with the following attributes:
+        ///   - <see cref="AutoGeneratedClassAttribute"/>: Marks classes containing auto-generated extension methods.
+        ///   - <see cref="AutoGeneratedHostBuilderConfigureMethodAttribute"/>: Marks auto-generated extension methods 
+        ///     within the classes that configure the <see cref="IHostBuilder"/>.
+        /// Once identified, these extension methods are executed to configure the host builder.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHostBuilder"/> instance.</param>
+        public static IHostBuilder InvokeAutoGeneratedConfigureMethods(this IHostBuilder builder)
+        {
+            var autoGeneratedExtensionMethods = Assembly.GetEntryAssembly()!
+                .GetTypes().Where(t => t.IsDefined(typeof(AutoGeneratedClassAttribute), inherit: false))
+                .SelectMany(t => t.GetMethods(BindingFlags.Static | BindingFlags.Public | BindingFlags.DeclaredOnly))
+                .Where(IsAutoGeneratedExtensionMethodForHostBuilder);
+
+            // Invoke the extension methods
+            var parameterValues = new object[] { builder };
+            foreach (var method in autoGeneratedExtensionMethods)
+            {
+                method.Invoke(obj: null, parameters: parameterValues);
+            }
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Checks whether a given method is an extension method for <see cref="IHostBuilder"/> auto generated by functions sdk.
+        /// </summary>
+        /// <param name="method">The <see cref="MethodInfo"/> representing the method to be checked.</param>
+        /// <returns>
+        /// true if the method is an extension method for <see cref="IHostBuilder"/> and has
+        /// the <see cref="AutoGeneratedHostBuilderConfigureMethodAttribute"/> decoration on it
+        /// otherwise, false.
+        /// </returns>
+        private static bool IsAutoGeneratedExtensionMethodForHostBuilder(MethodInfo method)
+        {
+            return method.IsDefined(typeof(System.Runtime.CompilerServices.ExtensionAttribute), inherit: false)
+                   && method.IsDefined(typeof(AutoGeneratedHostBuilderConfigureMethodAttribute))
+                   && method.GetParameters().Length > 0
```

Comment:
should you check parameters has a length of exactly 1? And that is `IHostBuilder`?

Reply:
This is not valid anymore as we switched from this implementation.